### PR TITLE
🐛(courserun) fix empty date on opened course run

### PR DESCRIPTION
**Pull request by 2018D**
opened course run show '...' when date is empty

## Purpose
Help fix issue #2063

## Proposal

Verify if value is empty before displaying it

### CODE REVIEW

File: src/frontend/js/widgets/SyllabusCourseRunsList/components/SyllabusCourseRun/index.tsx [hunk 1]
```diff
@@ -42,6 +42,12 @@ const messages = defineMessages({
 const OpenedCourseRun = ({ courseRun }: { courseRun: CourseRun }) => {
   const formatDate = useDateFormat();
   const intl = useIntl();
+  const enrollmentStart = courseRun.enrollment_start
```
Comment (sampaccoud) [inline src/frontend/js/widgets/SyllabusCourseRunsList/components/SyllabusCourseRun/index.tsx:45]: I'm worried that these changes did not break any test :thinking: 
Can you start by writing a test that fails and then fix it with this code?

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 - Course details characteristics overflow issue
 - Map all richie course properties into `getCourseGlimpseProps` util
+- Opened Course Run show empty date as '...' once more
```
Comment (NathanVss) [inline CHANGELOG.md:27]:
```suggestion
- Opened Course Run show empty date as '...'
```

**Review by NathanVss — COMMENTED** [review]
I agree with @sampaccoud, just add a test and it should be alright. Just in case, I can do it if you want, just let me know.

**Comment by 2018D** [comment]
New test added to check for forever open course. Needed Nathan's commit to have open_ongoing courserun when end date and enrollment_end date are empty.

File: CHANGELOG.md (hunk 2 above)
Comment (2018D) [reply CHANGELOG.md]: Done

**Comment by NathanVss** [comment]
I just merged the other fix PR, feel free to request another review once it is ready :) !

**Comment by 2018D** [comment]
Ready for review 😄

File: src/frontend/js/widgets/SyllabusCourseRunsList/index.spec.tsx [hunk 3]
```diff
@@ -232,6 +232,36 @@ describe('<SyllabusCourseRunsList/>', () => {
...
+      },
+    );
+
+    expect(getHeaderContainer().querySelectorAll('.course-detail__run-descriptions').length).toBe(
+      1,
+    );
+    screen.getByText('From Dec 31, 2020 to ...');
+    screen.getByText('From Dec 01, 2020 to ...');
+  });
+
```
Comment (NathanVss) [inline src/frontend/js/widgets/SyllabusCourseRunsList/index.spec.tsx:264]:
```suggestion
  it('has one forever open course run', async () => {
    const course = CourseLightFactory().one();
    const startDate = faker.date.past();
    const enrollmentStartDate = faker.date.past();
    const courseRun = CourseRunFactoryFromPriority(Priority.ONGOING_OPEN)({
      start: startDate.toISOString(),
      end: undefined,
      enrollment_start: enrollmentStartDate.toISOString(),
      enrollment_end: undefined,
      resource_link: 'https://openedx.endpoint/course-v1:edX+DemoX+Session1/info/',
    }).one();

    render(
      <SyllabusCourseRunsList
        courseRuns={[courseRun]}
        course={course}
        maxArchivedCourseRuns={MAX_ARCHIVED_COURSE_RUNS}
      />,
      {
        wrapper: Wrapper,
      },
    );

    expect(getHeaderContainer().querySelectorAll('.course-detail__run-descriptions').length).toBe(
      1,
    );
    getByRole(getHeaderContainer(), 'heading', {
      name: courseRun.title,
    });
    // Make sure that CourseRunEnrollment is well rendered.
    getByRole(getHeaderContainer(), 'button', { name: 'Log in to enroll' });

    const intl = createIntl({ locale: 'en' });
    getByText(
      getHeaderContainer(),
      'From ' + intl.formatDate(enrollmentStartDate, DEFAULT_DATE_FORMAT) + ' to ...',
    );
    getByText(
      getHeaderContainer(),
      'From ' + intl.formatDate(startDate, DEFAULT_DATE_FORMAT) + ' to ...',
    );
  });
```

**Review by NathanVss — COMMENTED** [review]
Great, I you could just update the test that way so it is more generic and homogeneous with existing tests and I will approve your PR, thank you 🙏

**Review by NathanVss — APPROVED** [review]

**Comment by NathanVss** [comment]
All good, @2018D it just needs a rebase in order to be merged. I tried to do it by myself but I don't have the permission to do it on your branch so you will have to do it 😅

**Comment by 2018D** [comment]
Rebase completed

**Pull request merged**